apps/docs/docs/ref/javascript/typescript-support.mdx

```diff
@@ -181,8 +219,20 @@ You can use the following helper types to make the TypeScript support easier to
     ```
 
     ```ts
-    const query = supabase.from('movies').select(`id, title`)
-    const movies: DbResult<typeof query> = await query
+    const countriesWithCitiesQuery = supabase.from("countries").select(
+      `
+      id,
+      name,
+      cities ( id, name )
+    `
+    );
+    type CountriesWithCities = DbResultOk<typeof countriesWithCitiesQuery>;
+
+    const {
+      data,
+      error,
+    }: { data: CountriesWithCities | null; error: DbResultErr | null } =
+      await countriesWithCitiesQuery;
```

Comment:
We should let this be inferred - `{ data: CountriesWithCities | null; error: DbResultErr | null }` needs more effort to satisfy type checks since it can include invalid states like `{ data: CountriesWithCities; error: DbResultErr }`. E.g. instead of doing:

```ts
if (error) throw error;
// data: DbResultOk
// ...do things with data...
```

You'd need:
```ts
if (error) throw error;
// data: DbResultOk | null
if (!data) throw new Error('...');
// data: DbResultOk
// ...do things with data...
```